### ISSUE TITLE
Unable to write logs using pipeline to database besides "public"

### ISSUE THREAD

**Issue by advayakrishna** [GreptimeTeam/greptimedb#4892]
### What type of bug is this?

Unexpected error

### What subsystems are affected?

Distributed Cluster, Frontend

### Minimal reproduce step

1. Create a pipeline definition
2. Try to write to pipeline with db query parameter set to something that is not "public"
3. Receive pipeline not found error

I was able to get around this by creating the pipeline with the header  "x-greptime-db-name: $dbname"
Then I see the pipeline in the greptime_private.pipelines table inserted with the schema column set to $dbname
However, then I receive an error like this:
{"code":1004,"error":"Failed to find schema, schema info: $dbname","execution_time_ms":726}

Writing the same log to the pipeline with db "public" works correctly and creates the table.

I suspect there is some internal pipeline handling logic related to schema creation that is not wired up correctly when the pipeline is created with a database that is not public.

### What did you expect to see?

Either I could reuse the same pipeline definition for different databases on write, or I should be able to create a pipeline for a different database then use that.

### What did you see instead?

1. Pipeline not found when writing to a database that is not "public"
2. Schema not found when forcing the pipeline "schema" column to be my target database

### What operating system did you use?

Helm Chart in Google Kubernetes Engine

### What version of GreptimeDB did you use?

0.9.3

**Comment by killme2008** [comment]
@paomian Please take a look

**Comment by paomian** [comment]
When you create a pipeline, it is created in the `public` database by default. You can also specify the database of the pipeline when creating the pipeline. The pipelines for the different db's are independent of each other.

```sql
mysql> select name,schema from greptime_private.pipelines;
+------+---------+
| name | schema  |
+------+---------+
| test | public  |
| test | test_db |
+------+---------+
2 rows in set (0.01 sec)
```

For example. If you want write log to `test_db` database. You can pass this db name via http url params.

```yaml
processors:
  - dissect:
      fields:
        - message
      patterns:
        - '%{ip_address} - - [%{timestamp}] "%{http_method} %{request_line}" %{status_code} %{response_size} "-" "%{user_agent}"'
      ignore_missing: true
  - date:
      fields:
        - timestamp
      formats:
        - "%d/%b/%Y:%H:%M:%S %z"

transform:
  - fields:
      - ip_address
      - http_method
    type: string
    index: tag
  - fields:
      - status_code
    type: int32
    index: tag
  - fields:
      - request_line
      - user_agent
    type: string
    index: fulltext
  - fields:
      - response_size
    type: int32
  - fields:
      - timestamp
    type: time
    index: timestamp
```

```bash
curl -X "POST" "http://localhost:4000/v1/events/pipelines/test?db=test_db" -F "file=@pipeline.yaml"
```

and then

```
curl -X "POST" "http://localhost:4000/v1/events/logs?db=test_db&table=pipeline_logs&pipeline_name=test" \
     -H 'Content-Type: application/json' \
     -d $'[
{"message":"127.0.0.1 - - [25/May/2024:20:16:37 +0000] \\"GET /index.html HTTP/1.1\\" 200 612 \\"-\\" \\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36\\""}
]'
```

you will write log success.
```
mysql> use test_db;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> show tables;
+---------------+
| Tables        |
+---------------+
| pipeline_logs |
+---------------+
1 row in set (0.02 sec)

mysql> select * from pipeline_logs;
+------------+-------------+-------------+----------------------+---------------------------------------------------------------------------------------------------------------------+---------------+---------------------+
| ip_address | http_method | status_code | request_line         | user_agent                                                                                                          | response_size | timestamp           |
+------------+-------------+-------------+----------------------+---------------------------------------------------------------------------------------------------------------------+---------------+---------------------+
| 127.0.0.1  | GET         |         200 | /index.html HTTP/1.1 | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36 |           612 | 2024-05-25 20:16:37 |
+------------+-------------+-------------+----------------------+---------------------------------------------------------------------------------------------------------------------+---------------+---------------------+
1 row in set (0.01 sec)
```

**Comment by advayakrishna** [comment]
Thanks for the info @paomian. I tried creating the pipeline with the `db` query variable and that succeeds (I see the pipeline in the `greptime_private.pipelines` table) but I am still hitting the schema error when trying to insert a row.

Same pipeline exists for both databases:
<img width="610" alt="image" src="https://github.com/user-attachments/assets/dcb102e3-9944-422d-81f7-d367b062f629">

Including logs from `greptime-frontend` where you can see the table get created for `db=public` and fail with the schema error for `db=errbor_interal`:

```
2024-10-29 09:19:14.058 PDT
2024-10-29T16:19:14.058575Z INFO pipeline::manager::table: Insert pipeline success, name: "agent_logs", table: "greptime.greptime_private.pipelines", output: Output { data: OutputData::AffectedRows(1), meta: OutputMeta { plan: None, cost: 0 } }
2024-10-29 09:19:47.349 PDT
2024-10-29T16:19:47.349243Z INFO operator::insert: Table `greptime.public.agent_logs` does not exist, try creating table
2024-10-29 09:19:47.584 PDT
2024-10-29T16:19:47.584016Z INFO create_table_inner: operator::statement::ddl: Successfully created table 'greptime.public.agent_logs' with table id 1080
2024-10-29 09:19:47.584 PDT
2024-10-29T16:19:47.584223Z INFO operator::insert: Successfully created table greptime.public.agent_logs with options: [("append_mode", "true")]
2024-10-29 09:20:50.508 PDT
2024-10-29T16:20:50.508692Z ERROR operator::insert: Failed to create table greptime.errbor_internal.agent_logs err=0: Failed to find schema, schema info: errbor_internal, at /greptimedb/src/operator/src/statement/ddl.rs:200:14
```

**Comment by paomian** [comment]
Can you provide information about writing logs? The Pipeline only associates a database name, it does not check if the database exists. The database cannot be created automatically, you need to create it manually, you need to use the following code to create the database

```
mysql> create database test_db;
Query OK, 1 row affected (0.00 sec)
```

I used the distributed version of 0.9.3, Here's how I did it.

1. start GreptimeDB cluster follow https://docs.greptime.cn/en/nightly/getting-started/installation/greptimedb-cluster#use-docker-compose
2. create databse `create database test_db;`
3. check database
```
mysql> show databases;
+--------------------+
| Database           |
+--------------------+
| greptime_private   |
| information_schema |
| pg_catalog         |
| public             |
+--------------------+
4 rows in set (0.01 sec)
```
4. create pipeline `curl -X "POST" "http://localhost:4000/v1/events/pipelines/test?db=test_db" -F "file=@pipeline.yaml"`
5. insert logs
```
curl -X "POST" "http://localhost:4000/v1/events/logs?db=test_db&table=pipeline_logs&pipeline_name=test" \
     -H 'Content-Type: application/json' \
     -d $'[{"message":"127.0.0.1 - - [25/May/2024:20:16:37 +0000] \\"GET /index.html HTTP/1.1\\" 200 612 \\"-\\" \\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36\\""}]'
```
6. query pipeline
```
mysql> select name,schema from greptime_private.pipelines;
+------+---------+
| name | schema  |
+------+---------+
| test | test_db |
+------+---------+
1 row in set (0.01 sec)
```
7. query logs
```
mysql> select * from test_db.pipeline_logs;
+------------+-------------+-------------+----------------------+---------------------------------------------------------------------------------------------------------------------+---------------+---------------------+
| ip_address | http_method | status_code | request_line         | user_agent                                                                                                          | response_size | timestamp           |
+------------+-------------+-------------+----------------------+---------------------------------------------------------------------------------------------------------------------+---------------+---------------------+
| 127.0.0.1  | GET         |         200 | /index.html HTTP/1.1 | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36 |           612 | 2024-05-25 20:16:37 |
| 127.0.0.1  | GET         |         200 | /index.html HTTP/1.1 | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36 |           612 | 2024-05-25 20:16:37 |
+------------+-------------+-------------+----------------------+---------------------------------------------------------------------------------------------------------------------+---------------+---------------------+
2 rows in set (0.00 sec)
```

My frontend only has the following two lines of logging
```
frontend0  | 2024-10-30T03:31:59.633032Z  INFO create_table_inner: operator::statement::ddl: Successfully created table 'greptime.test_db.pipeline_logs' with table id 1025
frontend0  | 2024-10-30T03:31:59.633055Z  INFO operator::insert: Successfully created table greptime.test_db.pipeline_logs with options: [("append_mode", "true")]
```

**Comment by advayakrishna** [comment]
Thank you for the detailed walkthrough, I went through your example on my instance and it worked.

I tried creating a new database for my pipeline and uploading it with that is also working.

Interestingly, I tried deleting the database "errbor_internal" I was using earlier and that gave me an error in the WebUI:
`Failed to find schema, schema info: errbor_internal`

Maybe something got corrupted with database in particular? Are there any queries to run or logs you would be interested in?

Also good to close because I'm not sure how to repro these conditions.
